File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        from collections import defaultdict
        d = defaultdict(int)
        d[0] = 1
        ans = 0
        rs = 0
        for i in nums:
            rs += i
            val = rs - k
            if val in d:
                ans += d[val]
            if rs in d:
                d[rs] += 1
            else:
                d[rs] = 1
        return ans

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int sum = 0, count = 0;
        Map<Integer, Integer> mp = new HashMap<>();
        mp.put(0,1);
        for(int i = 0; i < nums.length; i++){
            sum += nums[i];
            if(mp.containsKey(sum - k)){
                count += mp.get(sum - k);
            }
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }
        return count;
    }
}